core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -891,16 +896,13 @@ private[deploy] class Master(
     logInfo(s"Decommissioning the workers with host:ports ${workersToRemoveHostPorts}")
 
     // The workers are removed async to avoid blocking the receive loop for the entire batch
-    workersToRemove.foreach(wi => {
-      logInfo(s"Sending the worker decommission to ${wi.id} and ${wi.endpoint}")
-      self.send(WorkerDecommission(wi.id, wi.endpoint))
-    })
+    self.send(DecommissionWorkers(workersToRemove.map(_.id).toSeq))
 
     // Return the count of workers actually removed
     workersToRemove.size
   }
 
-  private def decommissionWorker(worker: WorkerInfo): Unit = {
+  private def decommissionWorker(worker: WorkerInfo, triggeredByWorker: Boolean): Unit = {
```

Comment:
There is only one place calling this method with `triggeredByWorker=false`. I think we can keep this method unchanged, and in that caller side:
```
case DecommissionWorkers(ids) =>
  ids.foreach ( id =>
    // We use foreach since get gives us an option and we can skip the failures.
    idToWorker.get(id).foreach { w =>
      decommissionWorker(w, triggeredByWorker = false))
      // Also send a message to the worker node to notify.
      worker.endpoint.send(DecommissionWorker)
    }
  )
```

Reply:
Sounds good.